scene/resources/curve.cpp

```diff
@@ -431,7 +441,9 @@ void Curve::set_data(const Array p_input) {
 	}
 
 	mark_dirty();
-	notify_property_list_changed();
+	if (old_size != new_size) {
+		notify_property_list_changed();
+	}
```

Comment:
I wonder if this will work correctly if you set the data of the same size. It's possible that the existing properties won't update to their new states, though it's probably fine.

Reply:
It shouldn't depend on the `property_list_changed` signal if it's just updating state. It may be more dependent on the `changed` signal that will be emitted in `Curve::mark_dirty()`.

Reply:
Yeah, it shouldn't, it just might, given this is how it works now. Well, we'll know soon enough :)